spec/models/concerns/courses/edit_options_spec.rb

```diff
@@ -62,6 +63,39 @@
         ],
       )
     end
+
+    it "sorts the potentinal subject by their name" do
+      expect(secondary_course.potential_subjects).to eql(
+        [
+          find_or_create(:secondary_subject, :art_and_design),
+          find_or_create(:secondary_subject, :biology),
+          find_or_create(:secondary_subject, :business_studies),
+          find_or_create(:secondary_subject, :chemistry),
+          find_or_create(:secondary_subject, :citizenship),
+          find_or_create(:secondary_subject, :classics),
+          find_or_create(:secondary_subject, :communication_and_media_studies),
+          find_or_create(:secondary_subject, :computing),
+          find_or_create(:secondary_subject, :dance),
+          find_or_create(:secondary_subject, :design_and_technology),
+          find_or_create(:secondary_subject, :drama),
+          find_or_create(:secondary_subject, :economics),
+          find_or_create(:secondary_subject, :english),
+          find_or_create(:secondary_subject, :geography),
+          find_or_create(:secondary_subject, :health_and_social_care),
+          find_or_create(:secondary_subject, :history),
+          find_or_create(:secondary_subject, :mathematics),
+          find_or_create(:secondary_subject, :modern_languages),
+          find_or_create(:secondary_subject, :music),
+          find_or_create(:secondary_subject, :philosophy),
+          find_or_create(:secondary_subject, :physical_education),
+          find_or_create(:secondary_subject, :physics),
+          find_or_create(:secondary_subject, :psychology),
+          find_or_create(:secondary_subject, :religious_education),
+          find_or_create(:secondary_subject, :science),
+          find_or_create(:secondary_subject, :social_sciences),
```

Comment:
it's surprising to me to see `find_or_create` in the assertion, can we change that to something that would just `find` and would `raise` if it couldn't find it?

Reply:
second question, where does this list come from, is it a fragile test because it encodes the whole subject list again?

Reply:
It's a sorted list of all the current secondary subjects. So it would break if the subject list changes. That may well actually be desirable. I've refactored it to use find. 

I also realised it was sorting primary subjects too. Not sure if this was desirable behaviour or not so i've changed the logic.

Reply:
`find` should work fine due to memoisation of the subjects 👍